src/connectedk8s/azext_connectedk8s/_params.py

```diff
@@ -32,8 +32,12 @@ def load_arguments(self, _):
         c.argument('proxy_cert', options_list=['--proxy-cert'], arg_group='Proxy', type=file_type, completer=FilesCompleter(), help='Path to the certificate file for proxy')
         c.argument('distribution', options_list=['--distribution'], help='The Kubernetes distribution which will be running on this connected cluster.', arg_type=get_enum_type(Distribution_Enum_Values))
         c.argument('infrastructure', options_list=['--infrastructure'], help='The infrastructure on which the Kubernetes cluster represented by this connected cluster will be running on.', arg_type=get_enum_type(Infrastructure_Enum_Values))
-        c.argument('disable_auto_upgrade', options_list=['--disable-auto-upgrade'], action='store_true', help='Flag to disable auto upgrade of arc agents.')
+        c.argument('auto_upgrade', options_list=['--auto-upgrade'], help='Flag to disable auto upgrade of arc agents. This flag must be provided when using custom registry repository', arg_type=get_enum_type(["true", "false"]))
```

Comment:
If we are renaming this parameter and handling deprecation guidelines, then the description should read 'must be set to false' instead of 'must be provided'

Reply:
ack

Reply:
ack

Reply:
So, what did we decide here? Deprecating the --disable-auto-upgrade?

Reply:
@shashankbarsin Did you get a chance to look at this?